docs/decisions/0008-application_state_differenciation.rst

```diff
@@ -0,0 +1,104 @@
+8. Intentional separation between local application state and API data state using ``@tanstack/react-query``
+============================================================================================================
+
+Status
+******
+
+Accepted  (January 2024)
+
+Context
+*******
+
+The ``frontend-app-admin-portal`` currently implements ``@tanstack/react-query`` predominantly within the ``learner-credit-management`` feature. ``@tanstack/react-query`` is a tool used to manage server state in react applications.
+
+See `0006-tanstack-react-query ADR <https://github.com/openedx/frontend-app-admin-portal/blob/master/docs/decisions/0006-tanstack-react-query.rst>`_ for more information
+
+As part of its implementation, a pattern has emerged of API data from the react-query cache is being populated within context providers. This has resulted in the unintended consequence of page refreshes when the top-level component context data is updated with the latest data from the ``react-query`` cache.
```

Comment:
nit: `react-query` has backticks around it in some places, but not here. Should they be consistent?